lib/jekyll/frontmatter_defaults.rb

```diff
@@ -100,12 +100,27 @@ def applies?(scope, path, type)
     def applies_path?(scope, path)
       return true if !scope.key?("path") || scope["path"].empty?
 
-      scope_path = Pathname.new(scope["path"])
-      Pathname.new(sanitize_path(path)).ascend do |ascended_path|
-        if ascended_path.to_s == scope_path.to_s
+      sanitized_path = Pathname.new(sanitize_path(path))
+
+      site_path = Pathname.new(@site.source)
+      rel_scope_path = Pathname.new(scope["path"])
+      abs_scope_path = File.join(@site.source, rel_scope_path)
+      Dir.glob(abs_scope_path).each do |scope_path|
+        scope_path = Pathname.new(scope_path).relative_path_from site_path
+        return true if path_is_subpath?(sanitized_path, scope_path)
+      end
+
+      path_is_subpath?(sanitized_path, rel_scope_path)
+    end
+
+    def path_is_subpath?(path, parent_path)
```

Comment:
do you want to expose this method as a `public` method? or would it rather be a `private` helper method?

Reply:
@ashmaroli Isn't there a private scope starting right before `applies?`

Reply:
right.. my bad.. ignore..